File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> keyinput, vector<int> board) {
    vector<int> answer(2, 0);
    
    int maxX = board[0] / 2;
    int maxY = board[1] / 2;
    
    for (int i = 0; i < keyinput.size(); i++) {
        if (keyinput[i] == "up") {
            if (answer[1] + 1 <= maxY)
                answer[1] += 1;
        }
        else if (keyinput[i] == "down") {
            if (answer[1] - 1 >= -maxY)
                answer[1] -= 1;
        }
        else if (keyinput[i] == "left") {
            if (answer[0] - 1 >= -maxX)
                answer[0] -= 1;
        }
        else if (keyinput[i] == "right") {
            if (answer[0] + 1 <= maxX)
                answer[0] += 1;
        }
    }
    
    return answer;
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.cs
using System;

public class Solution {
    public int[] solution(string[] keyinput, int[] board) {
        int[] answer = new int[] {0, 0};
        int maxX = board[0] / 2;
        int maxY = board[1] / 2;

        foreach (string key in keyinput)
        {
            if (key == "up" && answer[1] < maxY) answer[1]++;
            else if (key == "down" && answer[1] > -maxY) answer[1]--;
            else if (key == "left" && answer[0] > -maxX) answer[0]--;
            else if (key == "right" && answer[0] < maxX) answer[0]++;
        }

        return answer;
    }
}